### PR TITLE
Consistently handle multi-axis positioning and labels

### PR DESCRIPTION
**Pull request by philippjfr**
Supersedes https://github.com/holoviz/holoviews/pull/5816

### CODE REVIEW

**Comment by jlstevens** [comment]
Thanks!

I think this is more consistent and we can now delete the old axis swapping code that I think is now dead code because `yaxis` is no longer propagated to the overlay level.

Would be good to get this fix into this release even though I don't think we can claim swapping axes is supported until labels are also fixed...

File: holoviews/plotting/bokeh/element.py [hunk 1]
```diff
@@ -839,7 +853,8 @@ def _update_plot(self, key, plot, element=None):
         Updates plot parameters on every frame
         """
         plot.update(**self._plot_properties(key, element))
-        self._update_labels(key, plot, element)
+        if not self.multi_y:
+            self._update_labels(key, plot, element)
```
Comment (jlstevens) [inline holoviews/plotting/bokeh/element.py:857]: We should open an issue about this detailing what would be required...

**Comment by jlstevens** [comment]
With the labels fixed, I can go update the docs to say the `yaxis` option (and the label/fontsize) options are now supported!

Other than some of the known limitations (streaming data, fancier range streams, no dynamic labels) I think this nicely rounds out multi axis support for this release.

File: holoviews/plotting/bokeh/element.py [hunk 2]
```diff
@@ -608,14 +629,20 @@ def _init_plot(self, key, element, plots, ranges=None):
             # are not really an issue
             warnings.simplefilter('ignore', UserWarning)
             fig = figure(title=title, **properties)
+        fig.xaxis[0].update(**axis_props['x'])
+        fig.yaxis[0].update(**axis_props['y'])
```
Comment (jlstevens) [inline holoviews/plotting/bokeh/element.py:633]: Shame the Bokeh API doesn't let you set this up once together with the figure...

File: holoviews/plotting/bokeh/element.py [hunk 3]
```diff
@@ -497,20 +499,28 @@ def _create_extra_axes(self, plots, subplots, element, ranges):
...
-            yd = el.get_dimension(1)
+        for el, sp in zip(element, self.subplots.values()):
+            ax_dims = sp._get_axis_dims(el)[:2]
+            if sp.invert_axes:
+                ax_dims[::-1]
+            yd = ax_dims[1]
             dimensions[yd.name] = yd
             opts = el.opts.get('plot', backend='bokeh').kwargs
-            if yd.name in yaxes:
+            if not isinstance(yd, Dimension) or yd.name in yaxes:
```
[the remark this answers is not in the document]
Comment (philippjfr) [reply holoviews/plotting/bokeh/element.py:509]: You have added only a subset of all the possible element types. In the multi-dimension case such as two-level Bar or Violin charts you'd get a list/tuple of dimensions and in the statistics element case you can end up with None.

File: holoviews/tests/ui/bokeh/test_callback.py [hunk 4]
```diff
@@ -154,8 +154,8 @@ def test_multi_axis_rangexy(page, port):
     page.mouse.up()
 
     expected_xrange = (-35.1063829787234, 63.89361702127659)
-    expected_yrange1 = (717.2448979591848, 6657.244897959185)
-    expected_yrange2 = (-4232.7551020408155, 1707.2448979591848)
+    expected_yrange1 = (613.2089552238797, 6553.208955223879)
+    expected_yrange2 = (-4336.7910447761205, 1603.2089552238776)
```
[the remark this answers is not in the document]
Comment (philippjfr) [reply holoviews/tests/ui/bokeh/test_callback.py:158]: Axis label font size changed reducing the plot area.

File: holoviews/plotting/bokeh/element.py [hunk 5]
```diff
@@ -437,12 +439,15 @@ def _axis_props(self, plots, subplots, element, ranges, pos, *, dim=None,
...
-            xlabel, ylabel, zlabel = self._get_axis_labels((None, None) if (dim is None) else dims)
-            if self.invert_axes:
-                xlabel, ylabel = ylabel, xlabel
-                if dims:
-                    dims = dims[:2][::-1]
-            axis_label = ylabel if pos else xlabel
+            if dim:
+                axis_label = str(dim)
+            else:
+                xlabel, ylabel, zlabel = self._get_axis_labels(dims)
```
Comment (jlstevens) [inline holoviews/plotting/bokeh/element.py:445]: Currently, this is failing when `dims` is `None`...

**Review by jlstevens — APPROVED** [review]
Happy to see this merged once some of the windows tests go green.

**Pull request merged**